sql-server-2016-fci-existing-vnet-and-ad/dsc/ConfigSFCI.ps1

```diff
@@ -0,0 +1,291 @@
+
+configuration ConfigSFCI
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [String]$DomainName,
+
+        [Parameter(Mandatory)]
+        [System.Management.Automation.PSCredential]$Admincreds,
+
+        [Parameter(Mandatory)]
+        [System.Management.Automation.PSCredential]$svcCreds,
+
+        [Parameter(Mandatory)]
+        [String]$ClusterName,
+
+        [Parameter(Mandatory)]
+        [String]$SQLClusterName,
+
+        [Parameter(Mandatory)]
+        [String]$vmNamePrefix,
+
+        [Parameter(Mandatory)]
+        [Int]$vmCount,
+
+        [Parameter(Mandatory)]
+        [Int]$vmDiskSize,
+
+        [Parameter(Mandatory)]
+        [String]$witnessStorageName,
+
+        [Parameter(Mandatory)]
+        [System.Management.Automation.PSCredential]$witnessStorageKey,
+
+        [Parameter(Mandatory)]
+        [String]$clusterIP,
+
+        [String]$DomainNetbiosName=(Get-NetBIOSName -DomainName $DomainName),
+
+        [Int]$RetryCount=20,
+        [Int]$RetryIntervalSec=30,
+        [string]$driveLetter = 'S',
+        [Int]$probePort=37000 
+
+    )
+
+    Import-DscResource -ModuleName xComputerManagement, xFailOverCluster, xActiveDirectory, xSOFS, xSQLServer, xPendingReboot,xNetworking
+ 
+    [System.Management.Automation.PSCredential]$DomainCreds = New-Object System.Management.Automation.PSCredential ("${DomainNetbiosName}\$($Admincreds.UserName)", $Admincreds.Password)
+    [System.Management.Automation.PSCredential]$DomainFQDNCreds = New-Object System.Management.Automation.PSCredential ("${DomainName}\$($Admincreds.UserName)", $Admincreds.Password)
+    [string]$AdminUserNames = "${DomainNetbiosName}\Domain Admins"
+    
+    [System.Management.Automation.PSCredential]$ServiceCreds = New-Object System.Management.Automation.PSCredential ("${DomainNetbiosName}\$($svcCreds.UserName)", $svcCreds.Password)
+    [System.Management.Automation.PSCredential]$ServiceFQDNCreds = New-Object System.Management.Automation.PSCredential ("${DomainName}\$($svcCreds.UserName)", $svcCreds.Password)
+    
+    [System.Collections.ArrayList]$Nodes=@()
+    For ($count=0; $count -lt $vmCount; $count++) {
+        $Nodes.Add($vmNamePrefix + $Count.ToString())
+    }
+
+    Node localhost
+    {
+
+        # Set LCM to reboot if needed
+        LocalConfigurationManager
+        {
+            DebugMode = "ForceModuleImport"
+            RebootNodeIfNeeded = $true
+        }
+        
+        WindowsFeature FC
+        {
+            Name = "Failover-Clustering"
+            Ensure = "Present"
+        }
+
+		WindowsFeature FailoverClusterTools 
```

Comment:
can you clean up formatting in the file - indentation is off in a number of places